cargo-shuttle/src/config.rs

```diff
@@ -294,8 +294,26 @@ impl RequestContext {
     }
 
     pub fn get_local_config(
-        project_args: &ProjectArgs,
+        project_args: &mut ProjectArgs
     ) -> Result<Config<LocalConfigManager, ProjectConfig>> {
+        let root_directory_path = find_root_directory(&project_args.working_directory);
+
+        // Iterates through the directories until it finds the Cargo.toml file (project root) and returns the path
+        pub fn find_root_directory(dir: &Path) -> Option<PathBuf> {
+            for ancestor in dir.ancestors() {
+                if ancestor.join("Cargo.toml").exists() {
+                    return Some(ancestor.to_path_buf());
+                }
+            }
+            None
+        }
```

Comment:
Maybe this function does not need to be nested inside `get_local_config`? Maybe it can go all the way at the bottom of `lib.rs` (see other comment for why `lib.rs`)

Reply:
It also does not need to be `pub`lic

Reply:
Moved to lib.rs, removed public.